### PR TITLE
chore: update c++ base images

### PR DESCRIPTION
**Pull request by jcferretti**

### CODE REVIEW

File: cpp-client/build-dependencies.sh [hunk 1]
```diff
@@ -562,8 +562,8 @@ if [ "$CLONE_PROTOBUF" = "yes" ]; then
   echo
   echo "*** Cloning protobuf"
   cd $SRC
-  # Previously used version: v3.21.2
-  git clone $GIT_FLAGS -b v25.3 --depth 1 "${GITHUB_BASE_URL}/protocolbuffers/protobuf.git"
+  # Previously used version: v25.4
```
Comment (kosak) [inline cpp-client/build-dependencies.sh:565]: 25.3?

**Review by kosak — DISMISSED** [review]

**Review by kosak — APPROVED** [review]

**Review by kosak — DISMISSED** [review]

File: cpp-client/build.gradle [hunk 2]
```diff
@@ -38,6 +38,9 @@ def compare = tasks.register('compareProtobuf', DiffTask) {
     }
     generateTask.set ':cpp-client:updateProtobuf'
 }
+// Need to disable until the proto base image can catch up to protobuf 28.2
+tasks.getByName('compareProtobuf').enabled = false
```
Comment (devinrsmith) [inline cpp-client/build.gradle:42]: I'm confused why this is necessary; do we not use the same code to generate the c++ protobuf that we use to compare it?

Comment (jcferretti) [reply cpp-client/build.gradle:42]: The proto targets generate code for all languages with the protoc image.  The C++ build has never used that output, there is a separate script

https://github.com/deephaven/deephaven-core/blob/main/proto/proto-backplane-grpc/src/main/proto/build-cpp-protos.sh

that is used to generate code under cpp-client/deephaven/proto.  This code in the cpo-client/build.gradle was added to force both to be consistent.  But given the current issues in protobuf versions across languages and the state of chaos the protubuf team created with their breaking changes, this is not advisable anymore.  It actually never was advisable; there is no reason to force different languages to use the same proto version.  We arguably should not be generating C++ code if it is not used, but forcing something generated and not used to be consistent with something that is used and works and pass tests is not the solution to that IMHO

**Review by kosak — APPROVED** [review]

File: cpp-client/build.gradle (hunk 2 above)
Comment (niloc132) [reply cpp-client/build.gradle:42]: Went back and forth with Cristian a bit on this, and my understanding is that we previously had two ways to generate cpp code from protos, and they happened to have the same output - proto/proto-backplane-grpc's gradle/docker wiring, and the build-cpp-protos.sh script.

The fact that they generated the same output was wonderful, but ultimately irrelevant - the build-cpp-protos.sh script is going to use the protoc in the cpp docker image, and so is "more" correct. 

My proposal is that this merge as-is, with a new issue to be filed to follow-up here to restore this task. Both the updateProto and compareProto tasks should instead read from build-cpp-protos.sh output in its own docker task, and the rest of the cpp codegen in proto/proto-backplane-grpc should be removed.

We can revert this later and return to proto/proto-backplane-grpc generating cpp bindings, but if we do, we should at the same time delete build-cpp-protos.sh so there is only one way to generate this code. I'm not sure this is a step worth taking, especially given the current protobuf versioning ... scheme.

**Review by niloc132 — APPROVED** [review]

**Review by rcaudy — APPROVED** [review]

**Pull request merged**